shared/testutil/state.go

```diff
@@ -8,34 +8,43 @@ import (
 	"github.com/prysmaticlabs/prysm/shared/params"
 )
 
-var st, _ = stateTrie.InitializeFromProtoUnsafe(&pb.BeaconState{
-	BlockRoots:                 filledByteSlice2D(params.BeaconConfig().SlotsPerHistoricalRoot, 32),
-	StateRoots:                 filledByteSlice2D(params.BeaconConfig().SlotsPerHistoricalRoot, 32),
-	Slashings:                  make([]uint64, params.BeaconConfig().EpochsPerSlashingsVector),
-	RandaoMixes:                filledByteSlice2D(params.BeaconConfig().EpochsPerHistoricalVector, 32),
-	Validators:                 make([]*ethpb.Validator, 0),
-	CurrentJustifiedCheckpoint: &ethpb.Checkpoint{Root: make([]byte, 32)},
-	Eth1Data: &ethpb.Eth1Data{
-		DepositRoot: make([]byte, 32),
-		BlockHash:   make([]byte, 32),
-	},
-	Fork: &pb.Fork{
-		PreviousVersion: make([]byte, 4),
-		CurrentVersion:  make([]byte, 4),
-	},
-	Eth1DataVotes:               make([]*ethpb.Eth1Data, 0),
-	HistoricalRoots:             make([][]byte, 0),
-	JustificationBits:           bitfield.Bitvector4{0x0},
-	FinalizedCheckpoint:         &ethpb.Checkpoint{Root: make([]byte, 32)},
-	LatestBlockHeader:           HydrateBeaconHeader(&ethpb.BeaconBlockHeader{}),
-	PreviousEpochAttestations:   make([]*pb.PendingAttestation, 0),
-	CurrentEpochAttestations:    make([]*pb.PendingAttestation, 0),
-	PreviousJustifiedCheckpoint: &ethpb.Checkpoint{Root: make([]byte, 32)},
-})
-
 // NewBeaconState creates a beacon state with minimum marshalable fields.
-func NewBeaconState() *stateTrie.BeaconState {
-	return st.Copy()
+func NewBeaconState(options ...func(state *pb.BeaconState)) (*stateTrie.BeaconState, error) {
+	seed := &pb.BeaconState{
+		BlockRoots:                 filledByteSlice2D(params.MainnetConfig().SlotsPerHistoricalRoot, 32),
+		StateRoots:                 filledByteSlice2D(params.MainnetConfig().SlotsPerHistoricalRoot, 32),
+		Slashings:                  make([]uint64, params.MainnetConfig().EpochsPerSlashingsVector),
+		RandaoMixes:                filledByteSlice2D(params.MainnetConfig().EpochsPerHistoricalVector, 32),
```

Comment:
This changed from `BeaconConfig` to `MainnetConfig`. This is because minimal values cannot be encoded properly when saving to the database (I guess it's an ssz thing). If other values are really required, they can be passed in through `...options`.